apps/forms/avoid-losing-form-data/src/app/ui/dialog.component.ts

```diff
@@ -14,11 +13,13 @@ import { ChangeDetectionStrategy, Component } from '@angular/core';
 
       <div class="mt-4 flex gap-2">
         <button
+          (click)="dialogRef.close(true)"
```

Comment:
You can simply use the directives `[matDialogClose]="true"` and `[matDialogClose]="false"`, so you don't need the dialogRef injection at all 🚀